lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -2,25 +2,56 @@ module Azure
   module Armrest
     # Base class for services that need to run in a resource group
     class ResourceGroupBasedService < ArmrestService
+      # Create a resource +name+ within the resource group +rgroup+, or the
+      # resource group that was specified in the configuration, along with
+      # a hash of appropriate +options+.
+      #
+      # Returns an instance of the object that was created if possible,
+      # otherwise nil is returned.
+      #
+      # Note that this is an asynchronous operation. You can check the current
+      # status of the resource by inspecting the :response_headers instance and
+      # polling either the :azure_asyncoperation or :location URL.
+      #
       def create(name, rgroup = configuration.resource_group, options = {})
         validate_resource_group(rgroup)
         validate_resource(name)
 
         url = build_url(rgroup, name)
         url = yield(url) || url if block_given?
         response = rest_put(url, options.to_json)
-        model_class.new(response) unless response.empty?
+
+        obj = nil
+
+        unless response.empty?
+          obj = model_class.new(response.body)
+          obj.response_headers = Azure::Armrest::ResponseHeaders.new(response.headers)
+        end
+
+        obj
       end
 
       alias update create
 
+      # List all resources within the resource group +rgroup+, or the
+      # resource group that was specified in the configuration.
+      #
+      # Returns an ArmrestCollection, with the response headers set
+      # for the operation as a whole.
+      #
       def list(rgroup = configuration.resource_group)
         validate_resource_group(rgroup)
 
         url = build_url(rgroup)
         url = yield(url) || url if block_given?
+
         response = rest_get(url)
```

Comment:
I'm actually not entirely sure that Azure sets this in a uniform manner, or for all resources. Consider the way tokens are parsed for insights versus storage.

Also, I still have this issue outstanding: https://github.com/ManageIQ/azure-armrest/issues/154

Anyway, I agree in principal, but think it should be a separate PR for now.